### PR TITLE
Enabling POD providers to enforce a Terms & Conditions

### PR DESCRIPTION
**Pull request by megoth**
1. Added T&C-steps to options when setting up config
2. Presenting a checkbox if enforceToc is true
3. Validating that the checkbox is marked upon creating a new account (if enforceToc is true)

I've also added [handlebars-helpers](https://github.com/helpers/handlebars-helpers), but right now it's only used one place, so I understand if I should revert it. But I think it can be a useful tool once our templates grow more complex.

Apart from that there are some unnecessary changes, i.e. cleaning up some code and introducing some async/await, that I also can revert if needed.

### CODE REVIEW

**Comment by megoth** [comment]
This is a fix for https://github.com/solid/node-solid-server/issues/799

File: default-views/account/register-form.hbs [hunk 1]
```diff
@@ -71,6 +71,15 @@
             </span>
           </div>
 
+          {{#if (and enforceToc tocUri)}}
+            <div class="checkbox">
+              <label>
+                <input type="checkbox" name="acceptToc">
+                I agree to the <a href="{{tocUri}}" target="_blank">Terms & Conditions</a> of this service
```
Comment (RubenVerborgh) [inline default-views/account/register-form.hbs]: HTML encode

File: lib/requests/create-account-request.js [hunk 2]
```diff
@@ -26,12 +26,16 @@ class CreateAccountRequest extends AuthRequest {
...
+   * @param [options.enforceToc] {boolean} Whether or not to enforce the service provider's T&C
+   * @param [options.tocUri] {string} URI to the service provider's T&C
+   * @param [options.acceptToc] {boolean} Whether or not user has accepted T&C
    */
   constructor (options) {
     super(options)
 
     this.username = options.username
     this.userAccount = options.userAccount
+    this.acceptToc = options.acceptToc === 'true'
```
Comment (RubenVerborgh) [inline lib/requests/create-account-request.js]: should that parsing be this constructor's job?

Comment (RubenVerborgh) [reply lib/requests/create-account-request.js]: Probably `fromParams` (even though the whole `fromParams` pattern all over is a disease)

File: lib/requests/create-account-request.js [hunk 3]
```diff
@@ -251,6 +262,12 @@ class CreateOidcAccountRequest extends CreateAccountRequest {
       error.statusCode = 400
       throw error
     }
+
+    if (this.enforceToc && !this.acceptToc) {
+      error = new Error('Accepting Terms & Conditions are required for this service')
```
Comment (RubenVerborgh) [inline lib/requests/create-account-request.js]: *is

File: lib/requests/create-account-request.js [hunk 4]
```diff
@@ -312,6 +330,12 @@ class CreateTlsAccountRequest extends CreateAccountRequest {
...
+      error.statusCode = 400
+      throw error
+    }
```
Comment (RubenVerborgh) [inline lib/requests/create-account-request.js]: Code duplication?

**Comment by megoth** [comment]
@rubensworks Then I think I've fixed all the nits ^_^ Does it look ok?

**Comment by rubensworks** [comment]
@megoth Wrong Ruben,
I think you meant @RubenVerborgh :-)

**Comment by megoth** [comment]
Hah, sorry about that, did it a little bit fast here =P 

@RubenVerborgh does the changes look good?

**Comment by RubenVerborgh** [comment]
@megoth You can make your own Snyk account to see it.

**Comment by RubenVerborgh** [comment]
Fine with me.

**Pull request merged**